tapchannel/aux_sweeper.go

```diff
@@ -0,0 +1,402 @@
+package tapchannel
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"sync/atomic"
+
+	"github.com/btcsuite/btcd/btcec/v2/schnorr"
+	"github.com/lightninglabs/taproot-assets/address"
+	"github.com/lightninglabs/taproot-assets/asset"
+	"github.com/lightninglabs/taproot-assets/fn"
+	"github.com/lightninglabs/taproot-assets/proof"
+	cmsg "github.com/lightninglabs/taproot-assets/tapchannelmsg"
+	"github.com/lightninglabs/taproot-assets/tapfreighter"
+	"github.com/lightninglabs/taproot-assets/tappsbt"
+	"github.com/lightninglabs/taproot-assets/tapsend"
+	lfn "github.com/lightningnetwork/lnd/fn"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/lnwallet"
+	"github.com/lightningnetwork/lnd/tlv"
+)
+
+// resolutionReqs...
+type resolutionReq struct {
+	req lnwallet.ResolutionReq
+
+	resp chan lfn.Result[tlv.Blob]
+}
+
+// AuxSweeperCfg...
+type AuxSweeperCfg struct {
+	// AddrBook...
+	AddrBook *address.Book
+
+	// ChainParams are the chain parameters of the network the signer is
+	// operating on.
+	ChainParams address.ChainParams
+
+	// Signer is the backing wallet that can sign virtual packets.
+	Signer VirtualPacketSigner
+}
+
+// AuxSweeper...
+type AuxSweeper struct {
+	started atomic.Bool
+	stopped atomic.Bool
+
+	resolutionReqs chan *resolutionReq
+
+	cfg *AuxSweeperCfg
+
+	quit chan struct{}
+	wg   sync.WaitGroup
+}
+
+// NewAuxSweeper...
+func NewAuxSweeper(cfg *AuxSweeperCfg) *AuxSweeper {
+	return &AuxSweeper{
+		resolutionReqs: make(chan *resolutionReq),
+		cfg:            cfg,
+		quit:           make(chan struct{}),
```

Comment:
Why not use the `ContextGuard`, with that we get the wait group, quit channel and all shutdown aware contexts for free.